codex/validation.nim

```diff
@@ -124,9 +124,11 @@ proc start*(validation: Validation) {.async.} =
   validation.proofTimeout = await validation.market.proofTimeout()
   await validation.subscribeSlotFilled()
   validation.running = validation.run()
+  asyncSpawn validation.running
 
 proc stop*(validation: Validation) {.async.} =
-  await validation.running.cancelAndWait()
+  if not validation.running.isNil and not validation.running.finished:
```

Comment:
Ugh, when does this happen?

Reply:
yes

Reply:
It's a good question. Th nil check prevents SIGSEGV and the `finished` check prevents Defects like "future was completed more than once". Both are a PITA to debug and these checks prevent that pain. However, if `validation.running` is nil, then we could raise a `Defect` imo, because something didn't quite go right and we're in a bad state. BUT, we're stopping the module, so do we really care if `validation.running` is already nil? Probably not.